engines/efh/utils.cpp

```diff
@@ -0,0 +1,302 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "common/system.h"
+#include "common/random.h"
+#include "efh/efh.h"
+
+namespace Efh {
+
+void EfhEngine::setDefaultNoteDuration() {
+	// Original implementation is based on the int1C, which is triggered at 18.2065Hz.
+	// Every 4 times, it sets a flag (thus, approx every 220ms)
+	// The function was checking the keyboard in a loop, incrementing a counter and setting the last character read
+	// The "_defaultNoteDuration" was then set to 7 times this counter
+	//
+	// No implementation required in ScummVM
+}
+
+void EfhEngine::decryptImpFile(bool techMapFl) {
+	debugC(1, kDebugUtils, "decryptImpFile %s", techMapFl ? "True" : "False");
+	uint16 counter = 0;
+	uint16 target;
+	uint8 *curPtr;
+
+	if (!techMapFl) {
+		_imp2PtrArray[counter++] = curPtr = _imp2;
+		target = 431;
+	} else {
+		_imp1PtrArray[counter++] = curPtr = _imp1;
+		target = 99;
+	}
+
+	do {
+		*curPtr = (*curPtr - 3) ^ 0xD7;
+		if (*curPtr == 0x40) {
+			curPtr += 3;
+			if (!techMapFl)
+				_imp2PtrArray[counter++] = curPtr;
+			else
+				_imp1PtrArray[counter++] = curPtr;
+		} else
+			++curPtr;
+	} while (*curPtr != 0x60 && counter <= target);
+
+#ifdef debug
```

Comment:
Same thing, consider hiding it behind `--dump-scripts` command line flag

Reply:
Done